src/Microsoft.AspNet.Razor/Parser/TagHelpers/TagHelperBlockRewriter.cs

```diff
@@ -363,6 +387,35 @@ private static Block RebuildCodeGenerators(Block block)
             return builder.Build();
         }
 
+        private static SourceLocation GetAttributeNameStartLocation(SyntaxTreeNode node)
+        {
+            var startOffset = node.Start;
+
+            while (node.IsBlock)
+            {
+                // We want to track the start of the parent node. Each iteration of this while loop we will dive deeper
+                // into the Block. Since the attribute name will always be located at .Children[0] we need to track the
+                // corresponding .Start location of the attribute names parent so we can offset its .Start based on the 
+                // leading whitespace.
```

Comment:
nit: leading periods on properties make me want to read this as four sentences; don't think the dots add value